File: lib/model/product.dart
class Product {
  final int id;
  final String name;
  final String category;
  final double cost;
  final int discountPct;
  final double rating;
  final String ecom;
  final String ecomLogo;
  final String imageUrl;
  final String websiteUrl;
  final bool inStock;

  Product({
    this.id,
    this.name,
    this.category,
    this.cost,
    this.discountPct,
    this.rating,
    this.ecom,
    this.ecomLogo,
    this.imageUrl,
    this.websiteUrl,
    this.inStock,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'] ?? '',
      category: json['category'] ?? '',
      cost: (json['cost'] ?? 0).toDouble(),
      discountPct: json['discount_pct'] ?? 0,
      rating: (json['rating'] ?? 0).toDouble(),
      ecom: json['ecom'] ?? '',
      ecomLogo: json['ecom_logo'] ?? '',
      imageUrl: json['image_url'] ?? '',
      websiteUrl: json['website_url'] ?? '',
      inStock: json['in_stock'] ?? false,
    );
  }
}

File: backend/admin_service/app/models/product.py
from pydantic import BaseModel


class ProductCreate(BaseModel):
    name: str
    category: str | None = None
    cost: float = 0.0
    discount_pct: int = 0
    rating: float = 0.0
    ecom: str | None = None
    ecom_logo: str | None = None
    image_url: str | None = None
    website_url: str | None = None
    in_stock: bool = True


class ProductUpdate(BaseModel):
    name: str | None = None
    category: str | None = None
    cost: float | None = None
    discount_pct: int | None = None
    rating: float | None = None
    ecom: str | None = None
    image_url: str | None = None
    website_url: str | None = None
    in_stock: bool | None = None


class ProductOut(BaseModel):
    id: int
    name: str
    category: str | None
    cost: float
    discount_pct: int
    rating: float
    ecom: str | None
    ecom_logo: str | None
    image_url: str | None
    website_url: str | None
    in_stock: bool
